tests/sentry/models/test_projectcounter.py

```diff
@@ -1,13 +1,13 @@
+import pytest
+
+from sentry import options
 from sentry.models import Counter
-from sentry.testutils import TestCase
 
 
-class ProjectCounterTest(TestCase):
-    def test_increment(self):
-        user = self.create_user()
-        org = self.create_organization(owner=user)
-        team = self.create_team(organization=org)
-        project = self.create_project(teams=[team])
+@pytest.mark.django_db
+@pytest.mark.parametrize("upsert_sample_rate", [0, 1])
+def test_increment(default_project, upsert_sample_rate):
+    options.set("store.projectcounter-modern-upsert-sample-rate", upsert_sample_rate)
```

Comment:
Does this have a potential of leaking the option changes to other tests?

Reply:
database is reset after every test and caches are flushed